object_db.go

```diff
@@ -29,41 +32,72 @@ type ObjectDatabase struct {
 
 	// temp directory, defaults to os.TempDir
 	tmp string
+
+	// objectFormat is the object format (hash algorithm)
+	objectFormat string
+}
+
+type options struct {
+	alternates   string
+	objectFormat string
+}
+
+type Option func(*options)
```

Comment:
We unfortunately need to instantiate a variable in Git LFS itself with the appropriate type, since we need to build a slice of options that we can pass in.  I did intend to adopt this suggestion, but then realized that my generic instantiation function in Git LFS was not going to work if I did.